components/daikin_s21/s21.cpp

```diff
@@ -419,38 +419,88 @@ void DaikinS21::tx_next() {
 
   // Apply any pending settings
   // Important to clear the activate flag here as another command can be queued while waiting for this one to complete
-  if (activate_climate) {
-    activate_climate = false;
-    tx_command = "D1";
-    payload[0] = (pending.mode == climate::CLIMATE_MODE_OFF) ? '0' : '1'; // power
-    payload[1] = climate_mode_to_daikin(pending.mode);
-    payload[2] = (static_cast<int16_t>(pending.setpoint) / 5) + 28;
-    payload[3] = static_cast<char>(pending.fan);
+  if (this->activate_climate) {
+    this->activate_climate = false;
+    this->tx_command = "D1";
+    payload[0] = (this->pending.mode == climate::CLIMATE_MODE_OFF) ? '0' : '1'; // power
+    payload[1] = climate_mode_to_daikin(this->pending.mode);
+    payload[2] = (static_cast<int16_t>(this->pending.setpoint) / 5) + 28;
+    payload[3] = static_cast<char>(this->pending.fan);
     this->serial.send_frame(tx_command, payload);
     return;
   }
 
   if (activate_swing_mode) {
     this->activate_swing_mode = false;
-    tx_command = "D5";
-    payload[0] = climate_swing_mode_to_daikin(pending.swing);
-    payload[1] = (pending.swing != climate::CLIMATE_SWING_OFF) ? '?' : '0';
+    this->tx_command = "D5";
+    payload[0] = climate_swing_mode_to_daikin(this->pending.swing);
+    payload[1] = (this->pending.swing != climate::CLIMATE_SWING_OFF) ? '?' : '0';
     payload[2] = '0';
     payload[3] = '0';
     this->serial.send_frame(tx_command, payload);
     return;
   }
+
+  if (activate_preset) {
+    // potentially a two stage operation -- first disabling the old, then enabling the new
```

Comment:
I don't know if this is necessary without hardware to test with. Does a new command of lesser precedence (Econo) override powerful?